### ISSUE TITLE
[a11y]: 

### ISSUE THREAD

**Issue by pnayek** [carbon-design-system/carbon#16089]
### Package

carbon-components, carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

"carbon-components": "10.58.13", "carbon-components-react": "7.59.19"

### Automated testing tool and ruleset

IBM Accessibility Checker tool - The element with role "button" contains descendants with implicit roles "button" which are ignored by browsers

### Description

The IBM Accessibility Checker tool reports this issue: `The element with role "button" contains descendants with implicit roles "button" which are ignored by browsers` when we are using a `TooltipIcon` element on a `span` inside a `button`. Basically, in the below image, the `button` is the parent element -> then the `tooltipicon` is used on time displayed via `span` (In the below picture, the time is 22 minutes). It makes a hierarchy of `Button` -> `Button` -> `span`.

<img width="1705" alt="image" src="https://github.com/carbon-design-system/carbon/assets/15056711/c601917f-3263-4eb2-8753-c6eab96408a9">

- Is this a known Carbon issue? We are using "carbon-components": "10.58.13", "carbon-components-react": "7.59.19".
- Is there a best practice that needs to be followed to get past this issue?

### WCAG 2.1 Violation

4.1.2 Name, Role, Value

### Reproduction/example

N/A

### Steps to reproduce

N/A

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
You can't author a button within a button like that. You'll need to move things around in the DOM so the tooltip trigger isn't a child of the button. Usually we approach these situations by rendering the tooltip trigger (the time span in your case) adjacent to the button and then absolutely position it atop the button so it visually looks like it's part of that element, but isn't actually a child of it.

Another option would be to modify your design so that each item in the list that is currently a button could be modified to not be a button, and provide a separate button enabling you to drill into the item.

So right now you have something like this:

```
<ListItemButton>
  some content
  <TimeTooltip/>
</ListItemButton>  
```
You could instead shoot for something like this:

```
<ListItem>
  my content
  <TimeTooltip/>
  <ButtonToDrillIntoListItem> // this would handle the action/link that was previously on the entire list item
    <ArrowRight/> // optionally could include an icon like this or something
  </ButtonToDrillIntoListItem>
</ListItem>
```

I hope this helps. I don't think there's any actionable piece here on our end to fix so I'll close this out.
